Siren/Siren.swift

```
@@ -267,18 +279,30 @@ public class Siren: NSObject
         let session = NSURLSession.sharedSession()
         let task = session.dataTaskWithRequest(request, completionHandler: { (data, response, error) -> Void in
             
-            if data.length > 0 {
-                
-                // Convert JSON data to Swift Dictionary of type [String : AnyObject]
-                let appData = NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions.AllowFragments, error: nil) as? [String: AnyObject]
+            guard let error = error else {
+                guard let data = data else {
+                    if self.debugEnabled {
+                        print("[Siren] Error retrieving App Store data as no data was returned.")
+                    }
+                    return
+                }
                 
-                if let appData = appData {
+                // Convert JSON data to Swift Dictionary of type [String: AnyObject]
+                do {
+                    let jsonData = try NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions.AllowFragments)
+                    
+                    guard let appData = jsonData as? [String: AnyObject] else {
+                        if self.debugEnabled {
+                            print("[Siren] Error parsing App Store JSON data.")
+                        }
+                        return
```

Since this is in a `do…catch` construction, would it make sense to throw an error here instead of simply `return`?